src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.ByteArray.cs

```diff
@@ -108,24 +108,37 @@ public static byte[] SerializeToUtf8Bytes(object? value, Type inputType, JsonSer
             }
 
             Type runtimeType = GetRuntimeTypeAndValidateInputType(value, inputType);
-            return WriteCoreBytes(value!, GetTypeInfo(context, runtimeType));
+            return WriteBytesUsingGeneratedSerializer(value!, GetTypeInfo(context, runtimeType));
         }
 
         [RequiresUnreferencedCode(SerializationUnreferencedCodeMessage)]
         private static byte[] WriteCoreBytes<TValue>(in TValue value, Type runtimeType, JsonSerializerOptions? options)
         {
-            JsonTypeInfo jsonTypeInfo = GetTypeInfo(runtimeType, options);
-            return WriteCoreBytes(value, jsonTypeInfo);
+            JsonTypeInfo jsonTypeInfo = GetTypeInfo(options, runtimeType);
+            return WriteBytesUsingSerializer(value, jsonTypeInfo);
         }
 
-        private static byte[] WriteCoreBytes<TValue>(in TValue value, JsonTypeInfo jsonTypeInfo)
+        private static byte[] WriteBytesUsingGeneratedSerializer<TValue>(in TValue value, JsonTypeInfo jsonTypeInfo)
         {
             JsonSerializerOptions options = jsonTypeInfo.Options;
 
             using var output = new PooledByteBufferWriter(options.DefaultBufferSize);
             using (var writer = new Utf8JsonWriter(output, options.GetWriterOptions()))
```

Comment:
```suggestion
            using var writer = new Utf8JsonWriter(output, options.GetWriterOptions());
```

Reply:
For these cases, `writer` can be disposed before the end of the method, so I prefer the explicit scoping.

Reply:
Oh, I see now. Hadn't realized this changes semantics.